src/main/java/com/lginterview/filter/ThrottlingFilter.java

```diff
@@ -0,0 +1,67 @@
+package com.lginterview.filter;
+
+import com.google.common.util.concurrent.RateLimiter;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class ThrottlingFilter implements Filter {
+
+    private final String limitationPath;
+    private final RateLimiter rateLimiter;
+    private final String limitMessage;
+
+    @Autowired
+    public ThrottlingFilter(@Value("${courseservice.limitation-path}") String limitationPath,
+                            @Value("${courseservice.limitation-count}") int limitationCount,
+                            @Value("${courseservice.limitation-message}") String limitMessage) {
+        this.limitationPath = limitationPath;
+        this.rateLimiter = RateLimiter.create(limitationCount);
+        this.limitMessage = limitMessage;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+
+        if (!(servletRequest instanceof HttpServletRequest) ||
+                !(servletResponse instanceof HttpServletResponse)) {
+            throw new ServletException("Only http requests are supported ");
+        }
+
+        internalDoFilter((HttpServletRequest) servletRequest, (HttpServletResponse) servletResponse, filterChain);
+    }
+
+    private void internalDoFilter(HttpServletRequest servletRequest,
+                                  HttpServletResponse servletResponse,
+                                  FilterChain filterChain)
+            throws IOException, ServletException {
+
+        if (servletRequest.getRequestURI().contains(limitationPath)) {
```

Comment:
[MS][Minor] There are couple drawbacks:
1. You probably should use not "contains" but "startsWith".
2. You should add debugging log statement, because on production there will be a day, that there will be a user/customer which will get 503's, I you want be able to t
3. I think you should retry couple times (probably 5-10 times) before rejecting the request. 
If not you will force all clients to implement retries, especially when many clients (like browser) perform many request concurrently.

4. Is this solution better than actually setting a Tomcat to not accept more than X connections or not to have more than Y worker threads?

Reply:
Ad 1. Request Uri will also contain tomcat context-path setting. Changing it would require also changing filter setting but i agree.
Ad 2,3 I understand
Ad 4 I explained it in earlier comment